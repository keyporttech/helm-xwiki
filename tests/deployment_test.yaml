suite: test deployment
templates:
  - deployment.yaml
tests:
  - it: should use custom image
    set:
      image.tag: latest
      mysql.enabled: false
      postgresql.enabled: false
      image.name: default
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].image
          value: default:latest
      - matchSnapshot:
          path: spec.template.spec.containers[0].env
      - matchRegex:
          path: metadata.labels.app
          pattern: default-*
      - matchRegex:
          path: metadata.labels.app
          pattern: -latest          
  - it: should use mysql lts image
    set:
      image.tag: latest
      mysql.enabled: true
      postgresql.enabled: false
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].image
          value: xwiki:lts-mysql-tomcat
      - matchSnapshot:
          path: spec.template.spec.containers[0].env
      - matchRegex:
          path: metadata.labels.app
          pattern: xwiki-*
      - matchRegex:
          path: metadata.labels.app
          pattern: -lts-mysql-tomcat
  - it: should use postgres lts image
    set:
      image.tag: latest
      mysql.enabled: false
      postgresql.enabled: true
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].image
          value: xwiki:lts-postgres-tomcat
      - matchSnapshot:
          path: spec.template.spec.containers[0].env
      - matchRegex:
          path: metadata.labels.app
          pattern: xwiki-*
      - matchRegex:
          path: metadata.labels.app
          pattern: -lts-postgres-tomcat 
  - it: should properly set value
    set:
      image.pullPolicy: IfNotPresent
      service.internalPort: 8080
    asserts:  
      - equal:
          path: metadata.name
          value: RELEASE-NAME-xwiki
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: IfNotPresent
      - equal:
          path: spec.template.spec.containers[0].ports[0].containerPort
          value: 8080
      - equal:
          path: spec.template.spec.containers[0].volumeMounts[0].name
          value: xwiki-data
      - equal:
          path: spec.template.spec.volumes[0].name
          value: xwiki-data
      - equal:
          path: spec.template.spec.volumes[0].persistentVolumeClaim.claimName
          value: xwiki
      - matchSnapshot:
          path: spec.template.spec.containers[0].volumeMounts
      - matchSnapshot:
          path: spec.template.spec.volumes 
  - it: should properly set value pvc
    asserts:
      - isKind:
          of: PersistentVolumeClaim
        documentIndex: 1
      - matchSnapshot:
          path: spec
        documentIndex: 1
      - equal:
          path: metadata.name
          value: xwiki
        documentIndex: 1

          