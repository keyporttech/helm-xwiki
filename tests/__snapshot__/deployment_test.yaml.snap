should properly set value:
  1: |
    - mountPath: /opt/solr/server/solr
      name: xwiki-data
  2: |
    - name: xwiki-data
      persistentVolumeClaim:
        claimName: xwiki
should properly set value pvc:
  1: |
    accessModes:
    - ReadWriteMany
    resources:
      requests:
        storage: 500Mi
should use custom image:
  1: |
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: DB_PASSWORD
          name: RELEASE-NAME
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
should use mysql lts image:
  1: |
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: mysql-password
          name: RELEASE-NAME-mysql
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
should use postgres lts image:
  1: |
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: postgresql-password
          name: RELEASE-NAME-postgresql
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
